src/Query.php

```diff
@@ -103,20 +106,14 @@ public function __construct(Index $repository)
     }
 
     /**
-     * Adds fields to be selected from _source.
-     *
-     * Calling this function multiple times will append more fields to the
-     * list of fields to be selected from _source.
-     *
-     * If `true` is passed in the second argument, any previous selections
-     * will be overwritten with the list passed in the first argument.
-     *
-     * @param array $fields The list of fields to select from _source.
-     * @param bool $overwrite Whether or not to replace previous selections.
-     * @return $this
+     * @inheritDoc
      */
-    public function select($fields, bool $overwrite = false)
+    public function select(callable|array|string|float|int $fields, bool $overwrite = false)
     {
+        if (!is_array($fields)) {
+            throw new InvalidArgumentException('Unsuppoted type for $fields. Expected array.');
```

Comment:
Added exception.

Reply:
Yes. I was going to work on dropping QueryTrait dependencies and so probably this interface too.